cpp/ql/src/semmle/code/cpp/models/implementations/Sscanf.qll

```diff
@@ -0,0 +1,77 @@
+/**
+ * Provides implementation classes modeling `sscanf`, `fscanf` and various similar
+ * functions. See `semmle.code.cpp.models.Models` for usage information.
+ */
+
+import semmle.code.cpp.Function
+import semmle.code.cpp.commons.Scanf
+import semmle.code.cpp.models.interfaces.ArrayFunction
+import semmle.code.cpp.models.interfaces.Taint
+import semmle.code.cpp.models.interfaces.Alias
+import semmle.code.cpp.models.interfaces.SideEffect
+
+/**
+ * The standard function `sscanf`, `fscanf` and its assorted variants
+ */
+private class SscanfModel extends ArrayFunction, TaintFunction, AliasFunction, SideEffectFunction {
+  SscanfModel() { this instanceof Sscanf or this instanceof Fscanf or this instanceof Snscanf }
+
+  override predicate hasArrayWithNullTerminator(int bufParam) {
+    bufParam = this.(ScanfFunction).getFormatParameterIndex()
+    or
+    bufParam = this.(Sscanf).getInputParameterIndex()
```

Comment:
Or `(Snscanf).getInputParameterIndex()`?

Perhaps we should add a predicate to `ScanfFunction` that identifies whether the input is a string or stream so we don't need to special case this by QL class?

Reply:
Good point. Indeed, we should add a case for `Snscanf`.
> Perhaps we should add a predicate to ScanfFunction that identifies whether the input is a string or stream so we don't need to special case this by QL class?

That's a great idea! I'll prefer to just add the `Snscanf` disjunct in this PR and add a predicate to `ScanfFunction` as a follow-up PR, though.

Reply:
Sounds like a plan. :+1: